extension/extesionhelper/factory.go

```diff
@@ -0,0 +1,101 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensionhelper
+
+import (
+	"context"
+
+	"github.com/spf13/viper"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// FactoryOption apply changes to ExporterOptions.
+type FactoryOption func(o *factory)
+
+// CreateDefaultConfig is the equivalent of component.ExtensionFactory.CreateDefaultConfig()
+type CreateDefaultConfig func() configmodels.Extension
+
+// CreateTraceExtension is the equivalent of component.ExtensionFactory.CreateTraceExtension()
+type CreateServiceExtension func(context.Context, component.ExtensionCreateParams, configmodels.Extension) (component.ServiceExtension, error)
+
+// factory is the factory for Jaeger gRPC exporter.
+type factory struct {
+	cfgType                configmodels.Type
+	customUnmarshaler      component.CustomUnmarshaler
+	createDefaultConfig    CreateDefaultConfig
+	createServiceExtension CreateServiceExtension
+}
+
+// WithCustomUnmarshaler makes the factory to also implement component.ConfigUnmarshaler.
+func WithCustomUnmarshaler(customUnmarshaler component.CustomUnmarshaler) FactoryOption {
+	return func(o *factory) {
+		o.customUnmarshaler = customUnmarshaler
+	}
+}
+
+// NewFactory returns a component.ExtensionFactory that only supports all types.
```

Comment:
I don't think "only supports all types" is relevant for extensions?

Reply:
Done.